pkg/segment/results/mresults/metricresults.go

```diff
@@ -371,37 +366,26 @@ func (r *MetricsResult) GetResultsPromQl(mQuery *structs.MetricsQuery, pqlQueryt
 				err := errors.New("GetResultsPromQl: the length of tag key and tag value pair must match")
```

Comment:
Why are we making this check? The Tag Keys and the Tag Values may not be the same.
Or is there something I missed?

Reply:
Also, The Tag Keys is always unique, if it goes through the same Metrics Query Flow as SigLens. The `mQuery.ReorderTagFilters()` will make sure that the Tag Keys in the TagFilters are unique.
So you don't need `uniqueTagKeys` Map.